src/ir/item.rs

```diff
@@ -825,6 +826,50 @@ impl Item {
             _ => None,
         }
     }
+
+    /// Returns whether the item is a constified module enum
+    fn is_constified_enum_module(&self, ctx: &BindgenContext) -> bool {
```

Comment:
Here's what I came up with, I think this is what you're really trying to do with that function:

```rust
    /// Returns whether the item is a constified module enum.
    fn is_constified_enum_module(&self, ctx: &BindgenContext) -> bool {
        // Do not jump through aliases, except for aliases that point to a type
        // with the same name, since we don't generate code for them.
        let item = self.id.into_resolver().through_type_refs().resolve(ctx);
        let type_ = match *item.kind() {
            ItemKind::Type(ref type_) => type_,
            _ => return false,
        };

        match *type_.kind() {
            TypeKind::Enum(ref enum_) => {
                enum_.is_constified_enum_module(ctx, self)
            }
            TypeKind::Alias(inner) => {
                // TODO(emilio): Make this "hop through type aliases that aren't
                // really generated" an option in `ItemResolver`?
                let inner_item = ctx.resolve_item(inner);
                let name = item.canonical_name(ctx);

                if inner_item.canonical_name(ctx) != name {
                    return false;
                }

                return inner_item.is_constified_enum_module(ctx)
            }
            _ => false,
        }
    }
```